[Yeo Zhi Long] iP

yzhilong opened the pull request:
# Duke

> Some fancy quote

This **free** :grinning: program allows you to store information about the tasks you need to do. It can
* Save/load your task list to disc
* List your task list
* Add/remove tasks from your list
* Add a timestamp for your task

All you need to do is,
1. Download it from [here](https://github.com/yzhilong/ip)
2. Double click it
3. Learn the commands

Here is a sample command:
`deadline return book /by 22/9/2021 1800`


My program looks like this: 
```java
public class Duke {
    public static void main(String[] args) {
        ...
    }
}
```

Features to add
- [x] Save/Loading to disc
- [x] Gradle
- [ ] Week 4 onwards

Commit:
Add Gradle support

yzhilong committed:
Level 1

yzhilong committed:
Add Task.java

yzhilong committed:
Duke: change to switch statements

yzhilong committed:
Change Task constructor parameter

yzhilong committed:
Level 2

yzhilong committed:
Level 2

yzhilong committed:
add Task.java

yzhilong committed:
change to List<Task>

yzhilong committed:
Level 3

yzhilong committed:
Merge commit '60d82d6fc4887b9a4fa9d096e59d903730b4c64e'

# Conflicts:
#	src/main/java/Duke.java
#	src/main/java/Task.java

yzhilong committed:
add Deadline.java, Event.java, Todo.java

yzhilong committed:
add details to tasks

yzhilong committed:
shift logic into helper function

yzhilong committed:
add tabs to output

yzhilong committed:
fix: wrong Task type

yzhilong committed:
add: automated text UI testing

yzhilong committed:
add @override annotations to toString methods

yzhilong committed:
Add possible exceptions

yzhilong committed:
Remove duplicates

yzhilong committed:
Add InvalidTaskNumberException.java

yzhilong committed:
Add proper printout and more exceptions

yzhilong committed:
Level-5

yzhilong committed:
Change error message to fit deletions

yzhilong committed:
Level-6

yzhilong committed:
Refactor Tasks and Exceptions into separate packages

yzhilong committed:
Rename private attributes to make their purpose clearer

yzhilong committed:
Refactor parseInput using regex

yzhilong committed:
Add method to read data

yzhilong committed:
Add test data

yzhilong committed:
Add methods to output representation of task data

yzhilong committed:
Add Level-7

yzhilong committed:
Add sample data for saving and retrieving tasks

yzhilong committed:
Add Level-8

yzhilong committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/Tasks/Event.java

yzhilong committed:
Change how time data is saved

yzhilong committed:
Change how time data is stored in tasks

yzhilong committed:
Merge remote-tracking branch 'origin/Level-8'

# Conflicts:
#	src/main/java/Tasks/Deadline.java
#	src/main/java/Tasks/Event.java

yzhilong committed:
Fix error if we write empty task list to disk

yzhilong committed:
Add commands

yzhilong committed:
Refactor Duke into different classes

yzhilong committed:
Handle invalid timestamp bug

yzhilong committed:
Fix wrong data saved bug

yzhilong committed:
Remove show loading success printout

yzhilong committed:
Code cleanup

yzhilong committed:
Refactor into separate packages

yzhilong committed:
Add test data for storing/loading

yzhilong committed:
Rename data representation method

yzhilong committed:
Change Ui to handle inputs instead of Parser

yzhilong committed:
Add AddTaskCommand to handle when tasks are added

yzhilong committed:
Add String representations of updated classes

yzhilong committed:
Add JUnit tests for selected Command classes

yzhilong committed:
Add JUnit tests for selected utils

yzhilong committed:
Fix bug if data folder not present

yzhilong committed:
Update .gitignore

yzhilong committed:
Add find function to Duke

yzhilong committed:
Add JavaDocs

yzhilong committed:
Change code to follow coding standard

yzhilong committed:
Merge branch 'A-CodingStandard'

# Conflicts:
#	src/main/java/duke/commands/AddEventCommand.java
#	src/main/java/duke/commands/AddTodoCommand.java
#	src/main/java/duke/commands/ListCommand.java
#	src/main/java/duke/utils/Ui.java

yzhilong committed:
Merge tag 'Level-9'

# Conflicts:
#	src/main/java/duke/tasks/Task.java
#	src/main/java/duke/utils/Parser.java
#	src/main/java/duke/utils/TaskList.java

yzhilong committed:
Change printout when find is unable to find any tasks

yzhilong committed:
Remove unnecessary this keywords

weenleen commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,48 @@
+                ui.printOut(e.getMessage());
+            }
+        }
+        while (!isExited);
```
I think you should follow Egyptian style braces here!

chaiwanlin replied:
> I think you should follow Egyptian style braces here!
Agreed!

weenleen commented on src/main/java/duke/tasks/Task.java:
```diff
@@ -0,0 +1,47 @@
+        this.taskDetails = taskDetails;
+    }
+
+    public boolean markDone() {
```
Perhaps you could add a JavaDoc comment here, so that others will know what the returned boolean value represents

weenleen commented on src/main/java/duke/utils/Parser.java:
```diff
@@ -0,0 +1,73 @@
+        throw new UnknownCommandException();
+    }
+
+    private static String[] extractKeywordsFromCommand(String command, String[] keywords) {
```
The method name might be a bit lengthy. Maybe something like getKeywordsFrom?

weenleen commented on src/test/java/duke/commands/DeleteTaskCommandTest.java:
```diff
@@ -0,0 +1,40 @@
+    }
+
+    @Test
+    public void test0Index() {
```
Maybe this can be changed to follow the featureUnderTest_testScenario_expectedBehavior() format? Something like deleteCommand_zeroIndex_exceptionThrown()

weenleen reviewed (commented):
Good job! I think there aren't many coding standard violations :)

yzhilong commented:
# Duke

> Some fancy quote

This **free** :grinning: program allows you to store information about the tasks you need to do. It can
* Save/load your task list to disc
* List your task list
* Add/remove tasks from your list
* Add a timestamp for your task

All you need to do is,
1. Download it from [here](https://github.com/yzhilong/ip)
2. Double click it
3. Learn the commands

Here is a sample command:
`deadline return book /by 22/9/2021 1800`


My program looks like this: 
```java
public class Duke {
    public static void main(String[] args) {
        ...
    }
}
```

Features to add
- [x] Save/Loading to disc
- [x] Gradle
- [ ] Week 4 onwards

yzhilong committed:
Merge branch 'add-gradle-support' of https://github.com/yzhilong/ip into A-Gradle

yzhilong committed:
Fix wrong dates in test

chaiwanlin commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,48 @@
+    public void run() {
+        ui.showWelcome();
+        Command command;
+        boolean isExited = false;
```
Would it sound better if variable is named `isExit`?

chaiwanlin commented on src/main/java/duke/tasks/Deadline.java:
```diff
@@ -0,0 +1,37 @@
+
+    @Override
+    public String toString() {
+        String timeInfoString = timeInfo.format(outputFormatter);
```
I think just leaving at timeInfo will do since it's declared as String already and there are no other names in the method requiring the same name!

chaiwanlin commented on src/main/java/duke/utils/TaskList.java:
```diff
@@ -0,0 +1,129 @@
+     */
+    public String[] markDone(int i) throws InvalidTaskNumberException {
+        Task task = getTask(i);
+        boolean marked = task.markDone();
```
I think naming as 'isMarked' would be more in tune with the naming standard!

chaiwanlin reviewed (commented):
nit, otherwise LGTM. Great job! Code looks neat :+1:

yzhilong committed:
Fix wrong package name

yzhilong committed:
Merge remote-tracking branch 'origin/A-Gradle'

yzhilong committed:
Add checkstyle files

yzhilong committed:
Fix checkstyle violations in main

yzhilong committed:
Fix checkstyle violations in test

yzhilong committed:
Merge branch 'A-CheckStyle'

yzhilong committed:
Update build.gradle with javafx dependencies

yzhilong committed:
Add ignore checkStyle failures during build

yzhilong committed:
Add echo functionality to Duke

yzhilong committed:
Add String outputs to command printouts and executions

yzhilong committed:
Add proper duke responses

yzhilong committed:
Fix bug where duke does not close after "bye"

yzhilong committed:
Merge branch 'Level-10'

yzhilong committed:
Refactor Gui into separate class

yzhilong committed:
Fix display format and javadocs

yzhilong committed:
Rename Ui to CliUi

yzhilong committed:
Merge branch 'branch-Level-10'

yzhilong committed:
Change main class of build.gradle from Duke to Launcher

yzhilong committed:
Change pause duration to 2 seconds

yzhilong committed:
Change find method to "and" instead of "or" finding

yzhilong committed:
Add assertions

yzhilong committed:
Fix coding standard violations

yzhilong committed:
Merge pull request #1 from yzhilong/branch-A-Assertions

Add assertions

yzhilong committed:
Merge branch 'master' into branch-A-CodeQuality

yzhilong committed:
Merge pull request #2 from yzhilong/branch-A-CodeQuality

Fix coding standard violations

yzhilong committed:
Change DialogBox format

The current printout on the gui makes the user's input look similar
to Duke's output. This makes it seem like the user is chatting with
another person, but this is not the case. There is also no clear
demarcation of where the user's text is.

Changing the format of the output to include a clear textbox and
having Duke's response look different from the user's input will
make it clear that the user is chatting with a bot.

Let's add textboxes and change the color of the textboxes to
differentiate between the user's input and Duke's responses.

yzhilong committed:
Update Gui

yzhilong committed:
Refactor Gui

The Gui currently has multiple separate methods to create the elements
within it, which makes it a little hard for the user to follow.

Factoring out the setup into the constructor makes the
structure of the elements within Gui clearer.

yzhilong committed:
Add A-CI

yzhilong committed:
Add equals method to Tasks

yzhilong committed:
Add C-DetectDuplicates

yzhilong committed:
Fix done status affecting equality of Tasks

yzhilong committed:
Add JUnit tests for C-DetectDuplicates

yzhilong committed:
Merge branch 'branch-C-DetectDuplicates'

yzhilong committed:
Update README, Add Ui.png

yzhilong committed:
TaskList class: Add printout of how how many remaining tasks there are after delete command

yzhilong committed:
Fix java coding standards violations

yzhilong committed:
Add welcome message

yzhilong committed:
Fix java coding standard violations

yzhilong committed:
Add theme to user guide
